tests/stdlib/tos.nim

```diff
@@ -614,6 +614,10 @@ block osenv:
     doAssert existsEnv(dummyEnvVar) == false
     delEnv(dummyEnvVar)         # deleting an already deleted env var
     doAssert existsEnv(dummyEnvVar) == false
+  block putEnv:
```

Comment:
once the rest works (including windows), add tests for the issues mentioned in https://github.com/nim-lang/Nim/issues/18533; simplest is a separate module tests/stdlib/tosenv.nim with:
```nim
discard """
  matrix: "--threads"
  joinable: false
  target: "c cpp"
"""

```
...